packages/test-harness/package.json

```diff
@@ -24,11 +26,16 @@
   "dependencies": {
     "@cursorless/common": "workspace:*",
     "@cursorless/node-common": "workspace:*",
-    "glob": "^10.3.10"
+    "@cursorless/neovim-common": "workspace:*",
+    "@cursorless/neovim-registry": "workspace:*",
+    "glob": "^10.3.10",
+    "neovim": "5.1.0",
```

Comment:
Yes. `test-harness` is a neovim plugin, similar to `cursorless-neovim`. This is needed because `TestHarnessRun()` is called from `init.lua` (in both CI and locally), see `tests.md`. I am not sure what better solution you are thinking of?

Reply:
test-harness is itself not neovim-specific. We just pretend it's a neovim plugin. Ie we use the test harness for vscode, and many other test contexts. Eg the `index.ts` should really not be neovim-specific, because it is the entrypoint of this entire package which itself is used for testing all of our code. We should instead probably create a staging directory or something

Reply:
I know that test-harness is used for other contexts like vscode. We just don't use its entry point `index.ts` for other contexts, do we?

Reply:
Exactly. There's something awkward about a setup that is designed for a bunch of contexts but has an `index.ts` that only works for one

Reply:
Sure, just need to find a good name.

Atm we have:

* CI: `runNeovimTestsCI.ts` -> `launchNeovimAndRunTests.ts` -> `nvim` + `init.lua` -> `package.json` -> `index.ts`
* locally: `tasks.json` -> `nvim` + `init.lua` -> `package.json` -> `index.ts`

Reply:
Something like https://github.com/cursorless-dev/cursorless/pull/2256/commits/7eb7ee8871f20dfa0272a8b6653a98afa7b50377 is clearer I think. I would still argue for some sort of staging setup in the future, but for now at least it makes it clearer what's going on here

We could also drop a README here and mention this awkwardness in the README

Reply:
Yes, better. One thing worth mentioning is that the terminology for neovim is `plugin` instead of `extension` but I think that is still fine.

Reply:
feel free to tweak; I'm done for the night

Reply:
I'd be inclined to mark this conversation resolved and continue the discussion on #2564 but I'll leave it open in case you think this is something that should be addressed before this PR merges